### PR TITLE
Support vision model in chat app

### PR DESCRIPTION
**Pull request by rui-ren**
### Support vision model in ChatApp.

- [x] add the vision model support in Chat App example. 

image

### CODE REVIEW

File: examples/chat_app/interface/hddr_llm_onnx_interface.py
Comment (yufenglee) [inline examples/chat_app/interface/hddr_llm_onnx_interface.py:1]: why this file shows as empty?

File: examples/chat_app/assets/custom.css
Comment (yufenglee) [inline examples/chat_app/assets/custom.css:1]: and also for this file

**Review by natke — COMMENTED** [review]
PLease fix the codeql violations

Also we need to upgarde the gradio version to 4.14.0 due to a component governance violation

File: examples/chat_app/README.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ This is a chat demo using the various versions of the LLMs
 # set up
 
 1. Install **onnxruntime-genai-directml** 
+    > If you want to use CUDA model, you can download `onnxruntime-genai-cuda` package.
```
Comment (kunal-vaishnavi) [inline examples/chat_app/README.md:13]:
```suggestion
    > If you want to use a CUDA model, you can download the `onnxruntime-genai-cuda` package.
```

File: examples/chat_app/README.md [hunk 2]
```diff
@@ -30,6 +36,8 @@ This is a chat demo using the various versions of the LLMs
 
 2. Download models to the created folder, take phi-3-mini directml as example.
 
+   > For CUDA model, only need to create a new folder and download model inside, e.g. `microsoft/Phi-3-vision-128k-instruct-onnx-cuda`.
```
Comment (kunal-vaishnavi) [inline examples/chat_app/README.md:39]:
```suggestion
   > For a CUDA model, you only need to create a new folder and download the model inside (e.g. `microsoft/Phi-3-vision-128k-instruct-onnx-cuda`).
```

File: examples/chat_app/README.md [hunk 3]
```diff
@@ -43,6 +51,8 @@ This is a chat demo using the various versions of the LLMs
             --meta-llama_Llama-2-7b-chat-hf
             --mistralai_Mistral-7B-Instruct-v0.1
             ...
+        --cuda
+            --phi3-mini-v
```
Comment (kunal-vaishnavi) [inline examples/chat_app/README.md:55]:
```suggestion
            --phi-3-vision-128k-instruct
```

File: examples/chat_app/README.md [hunk 4]
```diff
@@ -62,4 +72,8 @@ This is a chat demo using the various versions of the LLMs
     ```
 
    Then open the local URL in broswer
-   ![alt text](image.png)
\ No newline at end of file
+   ![alt text](image.png)
+
+   For vision model, you will have blow UI interface.
```
Comment (kunal-vaishnavi) [inline examples/chat_app/README.md:77]:
```suggestion
   For a vision model, you will have the below UI interface.
```

File: examples/chat_app/interface/multimodal_onnx_interface.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+        prompt = self.template_header + prompt
+
+        prompt += f'{self.chat_template.format(input=text)}'
+        if len(prompt) > max_length:
+            history.clear()
+            prompt = f'{self.chat_template.format(input=text)}'
+
+        logging.info("Loading image ...")
+        # self.image = og.Images.open("/home/ruiren/Vision-Model-Demo/RING.jpg")
```
Comment (kunal-vaishnavi) [inline examples/chat_app/interface/multimodal_onnx_interface.py:40]:
```suggestion
```

**Review by natke — APPROVED** [review]

**Pull request merged**